File: examples/sample.py
def build_report(rows):
    # TODO: support grouped subtotals.
    return "\n".join(str(row) for row in rows)


def export_report(path, report):
    # FIXME: write atomically to avoid partial files.
    with open(path, "w", encoding="utf-8") as handle:
        handle.write(report)


LOREM = """
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.
Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed.
"""

File: examples/sample.ts
export function normalizeTitle(title: string): string {
  // TODO: preserve known acronyms when title-casing.
  return title.trim().toLowerCase();
}

export function parseDate(input: string): Date {
  // FIXME: reject invalid dates instead of relying on Date coercion.
  return new Date(input);
}

export const lorem = [
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
  "Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue.",
].join(" ");
